Classes/Connection/Elasticsearch/IndexFactory.php

```diff
@@ -60,12 +60,22 @@ public function __construct(ConfigurationContainerInterface $configuration)
         $this->configuration = $configuration;
     }
 
+    /**
+     * Get the index name from the typoscript settings.
+     *
+     * @return string
+     */
+    public function getIndexName()
+    {
+        return $this->configuration->get('connections.elasticsearch.index');
```

Comment:
This way it will be a breaking change.  An exception will be thrown on all existing installations.
I would recommend to use `return $this->configuration->getIfExists('connections.elasticsearch.index') ?: 'typo3content';`

Also another test is needed to check whether the default is returned. A unit test should be easy.